### PR TITLE
#1433: Multiple job files upload in monitor.

### PR DESCRIPTION
**Pull request by jhorcicka**
Solves #1433. 

In monitor, job file upload now supports multiple selection. 

file_upload

### CODE REVIEW

File: monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/JobsFolderController.java [hunk 1]
```diff
@@ -160,6 +161,29 @@
     }
 
     @RolesAllowed(SecurityRoles.JOB_EDITOR)
+    @RequestMapping(value = "/multiple_upload", method = RequestMethod.POST, produces = MediaType.TEXT_HTML_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public String uploadMultipleAnalysisJobsToFolderHtml(@PathVariable("tenant") final String tenant,
```
Comment (jhorcicka) [inline monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/JobsFolderController.java]: I created a new method with unique URL suffix "multiple_upload" not to change public API.

File: monitor/ui/src/main/webapp/scheduling.xhtml [hunk 2]
```diff
@@ -57,7 +55,7 @@
 						<span class="glyphicon glyphicon-plus" aria-hidden="true" /> <span>Build
 							job</span>
 					</button>
-					<form action="repository/#{user.tenant}/jobs/#{file.name}"
+					<form action="repository/#{user.tenant}/jobs/multiple_upload"
```
Comment (jhorcicka) [inline monitor/ui/src/main/webapp/scheduling.xhtml]: What does "#{file.name}" refer to? The value was (at least in my case) an empty string and I did not find any "File"Bean.java. Does it have any purpose or is/was that just some leftover?

Comment (LosD) [inline monitor/ui/src/main/webapp/scheduling.xhtml]: Good question. If it works without,  it probably was :-)

**Comment by LosD** [comment]
LGTM, but I can't test it right now. If you did a thorough test, merge when Travis is happy, otherwise I can do it later.

**Comment by jhorcicka** [comment]
I tested it on "/scheduling" (job file upload) but I am not sure if the xhtml template code is not used anywhere else. "conf.xml" file upload works fine (single selection). 

I am already using the feature on my localhost so my needs were satisfied. Now it can wait some time before it is merged into the master and another testing is always good. So, if you are willing to test, please do, @LosD. :-) Thanks.

**Comment by LosD** [comment]
Will do! :-)

**Comment by ClaudiaPHI** [comment]
LGTM!

**Comment by LosD** [comment]
Testing seems to work, but the URL looks weird, with a raw Java array reference in there: `http://localhost:8080/DataCleaner-monitor/scheduling?job_upload=Success&job_filename=[Ljava.lang.String;@fc18766`

We might want to get rid of that.

**Comment by LosD** [comment]
image
Picture version :)

**Comment by LosD** [comment]
Since you stopped using it for feedback, maybe just nuke it from the redirect in the multiple_upload endpoint (then you could maybe also check if it exists, and reinstate the filename uploaded if it does).

**Comment by jhorcicka** [comment]
Both, URL parameter job_filename and final success message now contain the list of files. 

file-upload

**Comment by LosD** [comment]
Uh, nice! Let's merge when Travis is happy

**Pull request merged**